resource-managers/mesos/src/main/scala/org/apache/spark/deploy/mesos/config.scala

```diff
@@ -101,10 +101,21 @@ package object config {
 
   private[spark] val executorSecretConfig = new MesosSecretConfig("executor")
 
+  private[spark] val CHECKPOINT =
```

Comment:
~small nit but since this is a boolean flag can we make it something like `USE_CHECKPOINTS` or `CHECKPOINT_ENABLED` for semantics?~

man... I was going to suggest this comment but then I see upstream created the APIs w/ just `checkpoint`. I guess you can disregard this.

Reply:
Struggled a bit with the naming too, but checked with the upstream